jerry-core/debugger/debugger-ws.c

```diff
@@ -433,16 +433,21 @@ JERRY_STATIC_ASSERT (JERRY_DEBUGGER_MAX_RECEIVE_SIZE < 126,
  *         false - otherwise
  */
 bool
-jerry_debugger_receive (void)
+jerry_debugger_receive (jerry_debugger_string_data_t **message_data_p) /**< custom message data */
 {
   JERRY_ASSERT (JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED);
 
+  if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_BREAKPOINT_MODE)
+      || (JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CLIENT_SOURCE_MODE))
+  {
+    JERRY_ASSERT (message_data_p != NULL);
```

Comment:
Another idea: we could create a define:

    #define JERRY_DEBUGGER_RECEIVE_DATA_MODE \
      (JERRY_DEBUGGER_BREAKPOINT_MODE | JERRY_DEBUGGER_CLIENT_SOURCE_MODE)

    JERRY_ASSERT (message_data_p != NULL
                  || (!(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_RECEIVE_DATA_MODE)

You can use a better symbol name of course.